lib/sla.js

```diff
@@ -61,35 +93,94 @@ function addChannelsToBridge(channel, bridge) {
  * Creates a channel to a specified endpoint and places it in the Stasis app.
  * @param {Object} client - Object that contains information from the ARI
  *   connection.
+ * @param {Object} inbound - Channel that originally dialed into Stasis.
  * @param {Object} bridge - Object that defines the bridge to be passed to
  *   addChannelsToBridge().
  * @return {Q} Q promise object.
  */
-function createChannel(client, bridge) {
+function originateChannel(client, inbound, bridge) {
   var defer = Q.defer();
-  var channel = client.Channel();
-  var originate = Q.denodeify(channel.originate.bind(channel));
-  originate({endpoint: 'SIP/phone', app: 'sla'})
-    .catch(function (err) {
-      defer.reject(err);
+  var dialed = [client.Channel(), client.Channel()];
+  var originate = [];
+  var numToHangup = dialed.length;
+  var station = ['SIP/phone1', 'SIP/phone2'];
+  dialed.forEach(function (channel, i, dialed) {
+    originate.push(Q.denodeify(dialed[i].originate.bind(dialed[i])));
+    originate[i]({endpoint: station[i], app: 'sla', appArgs: 'dialed',
+        timeout: 10})
+      .catch(function (err) {
+        defer.reject(err);
+      });
+    dialed[i].once('StasisStart', function(event, line) {
+      var answer = Q.denodeify(line.answer.bind(line));
+      answer();
+      var toKill = dialed.filter(function (unanswered) {
+        return unanswered.id !== line.id;
+      });
+      toKill.forEach(function (unanswered) {
+        var hangup = Q.denodeify(unanswered.hangup.bind(unanswered));
+        hangup();
+      });
+      defer.resolve(line);
+    });
+    inbound.once('ChannelHangupRequest', function(event, line) {
+      defer.resolve(null);
+      dialed.forEach(function (unanswered) {
+        var hangup = Q.denodeify(unanswered.hangup.bind(unanswered));
+        hangup();
+      });
+    });
+    dialed[i].once('ChannelDestroyed', function(event) {
+      numToHangup -= 1;
```

Comment:
You could also use a filter to remove the channel from the array and check for array.length === 0 instead of keeping track of a separate variable.